src/foraging_gui/Foraging.py

```diff
@@ -2191,26 +2206,34 @@ def _Start(self):
             # disable metadata fields
             self._set_metadata_enabled(False)
         else:
+            logging.info('Start button pressed: ending trial loop')
             self.Start.setStyleSheet("background-color : none")
             # enable metadata fields
             self._set_metadata_enabled(True)
-        # waiting for the finish of the last trial
-        if self.StartANewSession==1 and self.ANewTrial==0:
-            self.WarningLabel.setText('Waiting for the finish of the last trial!')
-            self.WarningLabel.setStyleSheet("color: red;")
-            while 1:
-                QApplication.processEvents()
-                if self.ANewTrial==1:
-                    self.WarningLabel.setText('')
-                    self.WarningLabel.setStyleSheet("color: red;")
-                    break
+            self._StopCurrentSession()  
```

Comment:
There is another constraint "self.StartANewSession==1" that is not contained in the self._StopCurrentSession(). We don't need to wait for the finish of the current trial if the Newsession button is not pressed. For example, the Start button is wrongly unpressed and we hope to press it again. 

I suggest we keep the original code.

Reply:
I'm not sure I understand your concern.
All "StopCurrentSession" does is toggle the start button color, and wait for the last trial to finish. This is exactly the logic that was implemented if the Start button was unchecked. 

If the user presses "start" on accident, they can still press "Start" again to resume trials. This code update doesn't impact that.

Reply:
If there is no constraint "self.StartANewSession==1", when you click the "Start" button when "self.ANewTrial==1", you will get an error like below:

![MicrosoftTeams-image (1)](https://github.com/AllenNeuralDynamics/dynamic-foraging-task/assets/109394934/003ed3e2-ce5d-4c62-9166-6f2a8c330ed8)

Reply:
We still need to add constraint "self.StartANewSession==1. 

The other modifications look good!